### PR TITLE
fix(pubsub): pass by reference explictly

### PR DESCRIPTION
**Pull request by alevenberg**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13220)

### CODE REVIEW

File: google/cloud/pubsub/internal/tracing_batch_sink_test.cc [hunk 1]
```diff
@@ -308,10 +308,11 @@ TEST(TracingBatchSink, AsyncPublishLargeBatch) {
   auto mock = std::make_shared<pubsub_testing::MockBatchSink>();
   EXPECT_CALL(*mock, AddMessage(_)).Times(batch_size);
   EXPECT_CALL(*mock, AsyncPublish)
-      .WillOnce([](google::pubsub::v1::PublishRequest const& request) {
-        EXPECT_THAT(request, IsProtoEqual(MakeRequest(batch_size)));
-        return make_ready_future(make_status_or(MakeResponse(request)));
-      });
+      .WillOnce(
+          [&batch_size](google::pubsub::v1::PublishRequest const& request) {
```
Comment (coryan) [inline google/cloud/pubsub/internal/tracing_batch_sink_test.cc:312]: `clang-tidy` seems unhappy. You could use `auto constexpr kBatchSize = kDefaultMaxLinks + 1;` above

**Review by coryan — APPROVED** [review]
I set a label to get full builds from Windows

File: google/cloud/pubsub/internal/tracing_batch_sink_test.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/pubsub/internal/tracing_batch_sink_test.cc:312]: I would also suggest capturing scalars by value (when not mutable), just like you would for a function parameters.

Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_batch_sink_test.cc]: Done, using the constexpr value

**Pull request merged**